src/Json/Decode/Applicative.elm

```diff
@@ -0,0 +1,19 @@
+module Json.Decode.Applicative exposing (apply, with)
+
+{-| Json.Decode.Applicative module
+
+This module implements typeclass applicative (apply : m (a -> b) -> a -> b)
+for Decoder functor based on the monadic return (succeed) and bind (andThen)
+
+-}
+
+import Json.Decode exposing (Decoder, andThen, succeed)
+
+
+apply : Decoder (a -> b) -> Decoder a -> Decoder b
+apply df da =
+    andThen (\a -> andThen (\f -> succeed (f a)) df) da
```

Comment:
this is fine but my favorite way f defining applicative map is `map2 (|>)` which is ridiculously simple.

Reply:
That is certainly more insightful.
I took this as an exercise of defining Applicative in terms of Monad and havent even thought of this neat way.